homeassistant/components/poolsense/__init__.py

```diff
@@ -41,9 +44,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
         _LOGGER.error("Invalid authentication")
         return False
 
-    coordinator = await get_coordinator(hass, entry)
+    coordinator = PoolSenseDataUpdateCoordinator(hass, entry)
```

Comment:
id recommend passing the login details to coordinator here rather than with each entity.

Reply:
Thanks. I have changed the structure to let the Poolsense class deal with the credentials as suggested.